server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -254,10 +253,8 @@ public SearchRequest(StreamInput in) throws IOException {
             finalReduce = true;
         }
         ccsMinimizeRoundtrips = in.readBoolean();
-        if ((in.getTransportVersion().before(TransportVersions.REMOVE_MIN_COMPATIBLE_SHARD_NODE)
-            || in.getTransportVersion().onOrAfter(TransportVersions.REVERT_REMOVE_MIN_COMPATIBLE_SHARD_NODE)) && in.readBoolean()) {
-            @UpdateForV9(owner = UpdateForV9.Owner.CORE_INFRA)  // this can be removed (again) when the v9 transport version can diverge
-            Version v = Version.readVersion(in); // and drop on the floor
+        if (in.getTransportVersion().before(TransportVersions.RE_REMOVE_MIN_COMPATIBLE_SHARD_NODE_90) && in.readBoolean()) {
```

Comment:
Maybe I am overthinking this, but shouldn't this be more complex?
IF 9.0 can talk with a ES version that is on a version with REVERT_REMOVE_MIN_COMPATIBLE_SHARD_NODE (8.18, unless this is backported to 8.18 too), it needs to take this branch.

So it could be (before(REMOVE_MIN_COMPATIBLE_SHARD_NODE) || onOrAfter(REVERT_REMOVE_MIN_COMPATIBLE_SHARD_NODE) &&  before(RE_REMOVE_MIN_COMPATIBLE_SHARD_NODE_90))?

Reply:
The "old behaviour" is before we removed min compatible, or after we reverted it but before we re-removed it

Reply:
I don't think so. The initial remove was immediately reverted before 8.17.3 and never released (except for maybe serverless), from a 9 perspective before `RE_REMOVE_MIN_COMPATIBLE_SHARD_NODE_90` the version always existed in the binary protocol.



```    
    public static final TransportVersion INITIAL_ELASTICSEARCH_8_16_6 = def(8_772_0_06);

    public static final TransportVersion REMOVE_MIN_COMPATIBLE_SHARD_NODE = def(8_773_0_00);
    public static final TransportVersion REVERT_REMOVE_MIN_COMPATIBLE_SHARD_NODE = def(8_774_0_00)

    public static final TransportVersion INITIAL_ELASTICSEARCH_8_17_3 = def(8_797_0_03);
```